asv_bench/benchmarks/benchmarks.py

```diff
@@ -819,4 +821,147 @@ def time_index(self, shape):
         return self.df.index
 
 
+class TimeIndexingNumericSeries:
+
+    param_names = ["shape", "dtype", "index_structure"]
+    params = [
+        get_benchmark_shapes("TimeIndexingNumericSeries"),
+        (np.int64, np.uint64, np.float64),
+        ("unique_monotonic_inc", "nonunique_monotonic_inc"),
+    ]
+
+    def setup(self, shape, dtype, index_structure):
+        N = shape[0]
+        indices = {
+            "unique_monotonic_inc": Index(range(N), dtype=dtype),
+            "nonunique_monotonic_inc": Index(
+                list(range(55)) + [54] + list(range(55, N - 1)), dtype=dtype
```

Comment:
Let's use not scalars, but some part of `N` (in all places where scalars are used)

Reply:
Replaced Scalars with parts of N.